openslides_votecollector/views.py

```diff
@@ -493,11 +505,16 @@ def post(self, request, poll_id, keypad_id):
             candidate = AssignmentOption.objects.filter(poll=poll_id).order_by('id').all()[key - 1].candidate
 
         # Save vote.
-        conn, created = AssignmentPollKeypadConnection.objects.get_or_create(poll=poll, keypad=keypad)
-        conn.serial_number = request.POST.get('sn')
-        conn.value = str(key)
-        conn.candidate = candidate
-        conn.save()
+        try:
+            conn = AssignmentPollKeypadConnection.objects.get(poll=poll, keypad=keypad)
+        except AssignmentPollKeypadConnection.DoesNotExist:
+            conn = AssignmentPollKeypadConnection()
+            conn.poll = poll
+            conn.keypad = keypad
+            conn.serial_number = request.POST.get('sn')
```

Comment:
You have to dedent these lines so that the new value for serial_number and value are saved also if then APKC already exists.